map-storage/src/Upload/S3FileSystem.ts

```diff
@@ -69,6 +70,19 @@ export class S3FileSystem implements FileSystemInterface {
         return;
     }
 
+    async listFiles(virtualDirectory: string, extension?: string): Promise<string[]> {
+        const list = await this.s3.listObjects({
+            Bucket: this.bucketName,
+        });
+        const recordsPaths: string[] = list.Contents?.map((record) => record.Key ?? "") ?? [];
+        if (extension) {
+            return recordsPaths
+                .filter((record) => path.extname(record) === extension)
+                .map((record) => path.relative(virtualDirectory, record));
+        }
+        return recordsPaths.map((record) => path.relative(virtualDirectory, record));
+    }
```

Comment:
This could potentially take a lot of time if there are many files (this would scan all files of ALL worlds in prod).
Also, I think it will fail because it does not implement pagination and S3 won't return more than 1000 records.

Take a look at the `deleteFiles` method.

It uses a "prefix" to scan files only in a given directory.
Also, it is handling pagination correctly I think (not sure that the s3.listObjects method is doing any pagination for you)